tests/base/develop_pipelines.py

```diff
@@ -97,7 +97,11 @@ def run_model_test(trainer_options, model, on_gpu: bool = True, version=None,
 
 
 def run_prediction(trained_model, dataloader, dp=False, min_acc=0.25):
-    if isinstance(trained_model, BoringModel):
+    ref_model = trained_model
+    if dp:
+        ref_model = trained_model.module
```

Comment:
problem is most likely here.
Perhaps it would be best to pass in `trainer.get_model()` to this function which will correctly unwrap the LightningModule.

Reply:
only one dp test is failing. `ref_model` is used only to check whether to go for `BoringModel` check function or `EvalModelTemplate`(the old one) check function. Also for dp `trainer.get_model() just returns `trained_model.module` only. There is some other issue here. Need to check.

Reply:
It's `BoringModel` for ref_model and `LightiningDataParallel` for trained_model.